src/Proto3/Wire/Decode.hs

```diff
@@ -393,16 +385,20 @@ byteString = bytes
 lazyByteString :: Parser RawPrimitive BL.ByteString
 lazyByteString = fmap BL.fromStrict bytes
 
--- | Parse a primitive with the @bytes@ wire type as 'Text'.
+-- | Decode a length-delimited wire field into a 'String'.
+string :: Parser RawPrimitive String
+string = Parser $ \case
+  LengthDelimitedField bs -> pure (ByteString.UTF8.toString bs)
```

Comment:
This will silently use replacement characters in place of invalid UTF-8.  It would be more consistent with existing practice, and perhaps slightly safer from a security perspective, if we were to fail out of the parser instead.